tool/tctl/common/terraform_command.go

```diff
@@ -0,0 +1,435 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/gravitational/teleport/api/types/common"
+	"io"
+	"log/slog"
+	"os"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/gravitational/teleport/api/constants"
+	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
+	machineidv1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/machineid/v1"
+	"github.com/gravitational/teleport/api/identityfile"
+	"github.com/gravitational/teleport/api/mfa"
+	"github.com/gravitational/teleport/api/types"
+	apiutils "github.com/gravitational/teleport/api/utils"
+	"github.com/gravitational/teleport/lib/auth/authclient"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+	"github.com/gravitational/teleport/lib/tbot"
+	"github.com/gravitational/teleport/lib/tbot/config"
+	"github.com/gravitational/teleport/lib/tbot/identity"
+	"github.com/gravitational/teleport/lib/tbot/ssh"
+	"github.com/gravitational/teleport/lib/tlsca"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+const (
+	terraformHelperDefaultResourcePrefix = "terraform-env-"
+	terraformHelperDefaultTTL            = "1h"
+)
+
+var terraformRoleSpec = types.RoleSpecV6{
```

Comment:
I'll move the rolespec there. The main question is: who should create the role?

If Teleport creates the role:
- this is easier, causes less updates
- this is not backward compatible
- users will see a new preset role appearing
- updating teleport will give you the new rules by default, no need to update the role

If tctl creates the role:
- you have the same version of the tctl you're running (for the best and the worst)
- you can have multiple tctl version fighting for the role content
- you can create multiple roles with different names
- you don't have this high-privilege role if you don't use it
- we cannot accidentally conflict with existing roles people created

As I'm wiring this, I'd rather have Teleport create the role as a preset. Some users might get ungappy with us adding more RBAC they don't use, but this will provide a better UX for most users.